src/index.php

```diff
@@ -1,18 +1,30 @@
 <?php
 require 'functions.php';
+require 'db_functions.php';
 
-echo produce_HTML_Head();
 $db = new PDO('mysql:host=db; dbname=cosmic_collector', 'root', 'password');
 $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
-$cosmic_event_query = $db->prepare('SELECT * FROM `cosmic_events`;');
-$cosmic_event_query->execute();
-$cosmic_events = $cosmic_event_query->fetchAll();
-
-echo '<div class="cosmic_events">';
+$cosmic_events = extract_cosmic_events($db);
+$elements_produced = [];
 foreach ($cosmic_events as $event) {
-	echo display_Cosmic_Event($event);
+    $event_elements = extract_event_elements($db, $event);
+    $elements_produced[$event['name']] = format_event_elements($event_elements);
```

Comment:
this is logic that hasn't been tested - should probably be a function so it can be unit tested